message/messaging_service.hh

```diff
@@ -341,9 +341,9 @@ public:
     future<> send_repair_put_row_diff(msg_addr id, uint32_t repair_meta_id, repair_rows_on_wire row_diff);
 
     // Wrapper for REPAIR_ROW_LEVEL_START
-    void register_repair_row_level_start(std::function<future<> (const rpc::client_info& cinfo, uint32_t repair_meta_id, sstring keyspace_name, sstring cf_name, dht::token_range range, row_level_diff_detect_algorithm algo, uint64_t max_row_buf_size, uint64_t seed, unsigned remote_shard, unsigned remote_shard_count, unsigned remote_ignore_msb, sstring remote_partitioner_name, table_schema_version schema_version)>&& func);
+    void register_repair_row_level_start(std::function<future<> (const rpc::client_info& cinfo, uint32_t repair_meta_id, sstring keyspace_name, sstring cf_name, dht::token_range range, row_level_diff_detect_algorithm algo, uint64_t max_row_buf_size, uint64_t seed, unsigned remote_shard, unsigned remote_shard_count, unsigned remote_ignore_msb, sstring remote_partitioner_name, table_schema_version schema_version, rpc::optional<streaming::stream_reason> reason)>&& func);
```

Comment:
What about idl/?

Reply:
rpc verb parameters are not part of the idl. rpc::optional is the way to extend parameters. We only append new parameters.

Reply:
What about enums? Shouldn't the enum be in idl? Or do we not support that either?

Reply:
We do have stream_reason in  idl/streaming.idl.hh. Why does it matter here? Before we do not send it at all, in the new version we send it. We are not extending existing parameter. 

Do you want to ask what happens if one of the existing parameter changes in idl? It works.

Reply:
@avikivity ^

Reply:
@avikivity `stream_reason` is already in `idl/streaming.idl.hh`.